### PR TITLE
Fix --string-non-empty option

### PR DESCRIPTION
**Pull request by romainbrenguier**
This was broken and the option had no effect

### CODE REVIEW

File: jbmc/src/java_bytecode/java_bytecode_language.cpp [hunk 1]
```diff
@@ -85,6 +85,8 @@ void java_bytecode_languaget::get_language_options(const cmdlinet &cmd)
     object_factory_parameters.max_nondet_string_length =
       safe_string2size_t(cmd.get_value("max-nondet-string-length"));
   }
+  if(cmd.isset("string-non-empty"))
+    object_factory_parameters.min_nondet_string_length = 1;
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/java_bytecode_language.cpp:72]: Can we please not have more command-line parsing in the language front-end? See
#2907 and #2908. Let alone doing so via an undocumented option?

**Review by tautschnig — CHANGES_REQUESTED** [review]
Let's please stop doing any command-line option parsing in the language front-end. Undocumented command-line options are a no-go.

File: jbmc/src/java_bytecode/object_factory_parameters.h [hunk 2]
```diff
@@ -28,6 +28,9 @@ struct object_factory_parameterst final
   /// Maximum value for the non-deterministically-chosen length of a string.
   size_t max_nondet_string_length=MAX_NONDET_STRING_LENGTH;
 
+  /// Maximum value for the non-deterministically-chosen length of a string.
```
Comment (thk123) [inline jbmc/src/java_bytecode/object_factory_parameters.h]: :no_entry_sign: :pick: Typo: `Minimum`

**Comment by romainbrenguier** [comment]
@thk123 @tautschnig this is ready for review

File: jbmc/regression/jbmc-strings/string-non-empty-option/test_non_empty.desc [hunk 3]
```diff
@@ -0,0 +1,9 @@
+CORE
+Test.class
+--function Test.checkMinLength --string-non-empty
+^EXIT=10$
+^SIGNAL=0$
+assertion.* line 11 function java::Test.checkMinLength.*: SUCCESS
```
Comment (thk123) [inline jbmc/regression/jbmc-strings/string-non-empty-option/test_non_empty.desc:6]: :no_entry_sign: This `desc` file is missing the line 17 assertion

**Review by thk123 — COMMENTED** [review]
Test is missing one assertion status - please add. Also, any reason to have a binary flag rather than just a `min-string-length` option?

File: jbmc/regression/jbmc-strings/string-non-empty-option/test.desc [hunk 4]
```diff
@@ -0,0 +1,10 @@
+CORE
+Test.class
+--function Test.checkMinLength
+^EXIT=10$
+^SIGNAL=0$
+assertion.* line 11 function java::Test.checkMinLength.*: FAILURE
+assertion.* line 13 function java::Test.checkMinLength.*: FAILURE
+assertion.* line 17 function java::Test.checkMinLength.*: FAILURE
+assertion.* line 19 function java::Test.checkMinLength.*: FAILURE
+
```
Comment (tautschnig) [inline jbmc/regression/jbmc-strings/string-non-empty-option/test.desc]: No blank lines here please, it confuses test runs on Windows. The same applies to the other *.desc files in this commit.

File: jbmc/src/java_bytecode/java_bytecode_language.cpp (hunk 1 above)
Comment (tautschnig) [reply jbmc/src/java_bytecode/java_bytecode_language.cpp:72]: While I remain unhappy it seems that #2908 is not currently progressing so I won't further block on this.

**Review by tautschnig — APPROVED** [review]
Ok from my point of view, except the *.desc files as noted below. The where-to-parse-command-line-options is postponed.

**Comment by romainbrenguier** [comment]
@thk123 
> Also, any reason to have a binary flag rather than just a min-string-length option?

No particular reason, I'm just fixing the previously existing feature that was broken, but it could be replaced by a min-string-length option in the future.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 0e36b4df0a9fb1376ea5689ab976a3af50d41d6b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86575976
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Review by thk123 — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 2d7e636b5efb991d7827a8479d030d7e2d85b89d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86578648
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 72350c2b5e131494d2c925a88b45921ae271d8a1).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86579860
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: cf123f06e879cc14ef87414a67c6308cf2f00a8c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86580449

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**